modules/internal/DefaultSparse.chpl

```diff
@@ -304,18 +310,17 @@ module DefaultSparse {
 
       //linearly fill the new colIdx from backwards
       var newIndIdx = indsDom.high; //index into new indices
-      var oldIndIdx = oldnnz; //index into old indices
+      var oldIndIdx = oldnnz-1; //index into old indices
       var newLoc = actualInsertPts[newIndIdx]; //its position-to-be in new dom
       while newLoc == -1 {
         newIndIdx -= 1;
         if newIndIdx == indsDom.low-1 then break; //there were duplicates -- now done
         newLoc = actualInsertPts[newIndIdx];
       }
+      var arrShiftMap: [{0..#oldnnz}] int; //to map where data goes
```

Comment:
Dropping the anonymous domain is preferred, and it doesn't hurt to clean up while we're here (though you can consider this feedback optional):

```suggestion
      var arrShiftMap: [0..#oldnnz] int; //to map where data goes
```

Reply:
Done in sparse-0based branch.